### PR TITLE
PUBDEV-8681 GAM predict failure with fold_column

### PR DESCRIPTION
**Pull request by wendycwong**
This PR fixes the bug in JIRA: https://h2oai.atlassian.net/browse/PUBDEV-8681.

There is a bug in my code where the fold column is mistreated as a normal column to score for. 

I added code to make sure the fold column is treated properly.

### CODE REVIEW

File: h2o-r/tests/testdir_algos/gam/runit_PUBDEV_8681_paul_gam_cv_no_cv.R [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        gam_columns = c("C4"),
+        scale = c(.05),
+        num_knots = c(5),
+        spline_orders = c(3)
+    )
+    test <- train[-c(6)]
+    
+    # no cross validation, bs = 0 (default)
+    mod <- do.call(what = "h2o.gam", args = params)
+    browser()
```
Comment (michalkurka) [inline h2o-r/tests/testdir_algos/gam/runit_PUBDEV_8681_paul_gam_cv_no_cv.R:32]: probably just forgotten

File: h2o-core/src/main/java/hex/ModelBuilder.java [hunk 2]
```diff
@@ -256,7 +256,12 @@ public void compute2() {
       } finally {
         _parms.read_unlock_frames(_job);
         if (!_parms._is_cv_model) cleanUp(); //cv calls cleanUp on its own terms
-        Scope.exit();
+        if (!_parms._is_cv_model && _parms._fold_column != null) {
```
Comment (michalkurka) [inline h2o-core/src/main/java/hex/ModelBuilder.java:259]: I think this really should not be on ModelBuilder - I will request changes based on these lines.

We can think of better solution together.

File: h2o-algos/src/main/java/hex/gam/MatrixFrameUtils/GamUtils.java [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package hex.gam.MatrixFrameUtils;
 
+import com.google.gson.internal.Streams;
```
Comment (michalkurka) [inline h2o-algos/src/main/java/hex/gam/MatrixFrameUtils/GamUtils.java:3]: looks unused

**Review by michalkurka — CHANGES_REQUESTED** [review]
Requesting changes due to specific handling of fold column in ModelBuilder - see inline comments

File: h2o-r/tests/testdir_algos/gam/runit_PUBDEV_8681_paul_gam_cv_no_cv.R (hunk 1 above)
Comment (wendycwong) [reply h2o-r/tests/testdir_algos/gam/runit_PUBDEV_8681_paul_gam_cv_no_cv.R]: Sorry.  Fixed.

File: h2o-algos/src/main/java/hex/gam/MatrixFrameUtils/GamUtils.java (hunk 3 above)
Comment (wendycwong) [reply h2o-algos/src/main/java/hex/gam/MatrixFrameUtils/GamUtils.java]: I removed all unused code.  I hope.

**Review by michalkurka — APPROVED** [review]
LGTM, thank you @wendycwong

**Pull request merged**